Makefile

```diff
@@ -324,6 +324,14 @@ include tools/afl/fuzzy.mk
 
 err: ; $(ERROR)
 
+fmt:
+	@git ls-files | egrep '.*\.(c|h|cpp|hpp)$$' | xargs clang-format -i
+
+fmt_check:
+	@git ls-files | egrep '.*\.(c|h|cpp|hpp)$$' | xargs clang-format -i
+	@git --no-pager diff
+	@test $$(git diff | head -c 1 | wc -c) = 0
+
```

Comment:
These targets operate incorrectly for two reasons:
- they operate on all code, but we have some third-party imported code. We should not touch the formatting of the imported code, because when you bump the version - it is very hard to review, because of a large diff.
- this is not very make-idiomatic

IMO the proper way would be to create make "patterned rules" (https://www.gnu.org/software/make/manual/html_node/Pattern-Rules.html) starting from `.c` files, but we can later add rules for other languages with good formatters like Go or Rust.

Then, actual format targets just depend on the implicit targets, created by those rules: we already define many variables in our makefiles, which have separate lists of our sources vs third-party sources, so it should be easy to do it.

If the above does not make sense, here's a simple example:
```
ignat@dev:~/formatme$ find ./
./
./mysrc
./mysrc/hello.c
./thirdparty
./thirdparty/donttouch.c
./Makefile
```

The makefile
```
#formatme
MY_SRC := $(wildcard mysrc/*.c)
THIRD_PARTY := $(wildcard thirdparty/*.c)

%.c.check-fmt: %.c
	clang-format $< | diff -q $< -

%.c.fmt: %.c
	clang-format -i $<

check-fmt: $(patsubst %.c,%.c.check-fmt,$(MY_SRC))
	echo checked

fmt: $(patsubst %.c,%.c.fmt,$(MY_SRC))
	echo formatted
```

the output
```
ignat@dev:~/formatme$ make
clang-format mysrc/hello.c | diff -q mysrc/hello.c -
Files mysrc/hello.c and - differ
Makefile:5: recipe for target 'mysrc/hello.c.check-fmt' failed
make: *** [mysrc/hello.c.check-fmt] Error 1
ignat@dev:~/formatme$ make fmt
clang-format -i mysrc/hello.c
echo formatted
formatted
ignat@dev:~/formatme$ make
clang-format mysrc/hello.c | diff -q mysrc/hello.c -
echo checked
checked
```

here, we ensured the formatting of our sources, but did not touch the thirdparty ones. Also, no "for" loops, xargs etc

Reply:
@secumod, thanks for pointing this out! I have come to a similar approach when trying to integrate clang-tidy (as it requires compilation flags to be available), so I guess that's the way to go 👍

I also don't really like that checking format requires **git** invocations as that would break _make_ usage with distributed source tarballs. This new approach should allow to avoid that.

Expect a reroll of this PR with the new approach (and integrated clang-tidy).